src/applications/bmqbrkr/bmqbrkr.m.cpp

```diff
@@ -672,6 +651,46 @@ int main(int argc, const char* argv[])
         return mqbu::ExitCode::e_COMMAND_LINE;  // RETURN
     }
 
+    if (version) {
+        bsl::cout << bmqbrkrscm::Version::version() << "\n";
+        return 0;
+    }
+
+    if (configDir.empty()) {
+        bsl::cerr << "Error: No value supplied for the non-option argument "
+                     "\"config\".\n";
+    }
+
+    printStartStopTrace("STARTING");
+
+    ignoreSigpipe();
+
+    // Register a SIGINT handler to allow graceful shutdown of the broker from
+    // lower layers.
+    struct sigaction sa;
+    sigemptyset(&sa.sa_mask);
+    sa.sa_flags   = 0;
+    sa.sa_handler = appShutdownSignal;
+
+    int rc = ::sigaction(SIGINT, &sa, NULL);
+    if (rc != 0) {
+        // Note that this is not fatal, as the primary way to stop the broker
+        // is not through a 'kill -s INT', but via issueing an 'exit' command
+        // through the control pipe.
+        bsl::cerr << "Failed to install SIGINT handler  (rc: " << rc << ")\n"
+                  << bsl::flush;
+    }
+
+    // If running in a TTY, enable signal handler so that user signals can be
+    // used to stop the task.
+    if (isatty(fileno(stdout))) {
+        ::sigaction(SIGQUIT, &sa, NULL);
+        ::sigaction(SIGTERM, &sa, NULL);
+    }
```

Comment:
Any reason to move the signal handling related logic here instead of keeping it earlier?

Reply:
We don't need to do that (nor to print "STARTING...")  if we are just printing the version.